bin/generate-php-sync-issue.mjs

```diff
@@ -0,0 +1,433 @@
+		);
+		process.exit( 1 );
+	}
+
+	const sinceArg = getArg( 'since' );
+	let since;
+
+	if ( sinceArg ) {
+		if ( validateSince( sinceArg ) ) {
+			since = sinceArg;
+		} else {
+			console.error(
+				'Error: The --since argument cannot be more than 2 months from the current date.'
+			);
+			process.exit( 1 );
+		}
+	} else {
+		console.error(
+			`Error. The --since argument is required (e.g. YYYY-MM-DD). This should be the date of the previous release's final RC.`
+		);
+		process.exit( 1 );
+	}
+
+	console.log( 'Welcome to the PHP Sync Issue Generator!' );
+
+	if ( DEBUG ) {
+		console.log( 'DEBUG MODE' );
+	}
+
+	const octokit = new Octokit( {
+		auth: authToken,
+	} );
+
+	console.log( '--------------------------------' );
+	console.log( '• Running script...' );
+
+	// These should be paths where we expect to find PHP files that
+	// will require syncing to WordPress Core. This list should be
+	// extremely selective.
+	const paths = [ '/lib', '/phpunit' ];
+
+	console.log( `• Fetching all commits made to ${ REPO } since: ${ since }` );
+	let commits = await fetchAllCommitsFromPaths( since, paths );
+
+	// Remove identical commits based on sha
+	commits = commits.reduce( ( acc, current ) => {
+		const x = acc.find( ( item ) => item.sha === current.sha );
+		if ( ! x ) {
+			return acc.concat( [ current ] );
+		}
+		return acc;
+	}, [] );
+
+	// Fetch the full commit data for each of the commits.
+	// This is because the /commits endpoint does not include the
+	// information about the `files` modified in the commit.
+	console.log(
+		`• Fetching full commit data for ${ commits.length } commits`
+	);
+	const commitsWithCommitData = await Promise.all(
+		commits.map( async ( commit ) => {
+			const commitData = await fetchCommit( commit.sha );
+
+			// Our Issue links to the PRs associated with the commits so we must
+			// provide this data. We could also get the PR data from the commit data,
+			// using getPullRequestDataForCommit, but that requires yet another
+			// network request. Therefore we optimise for trying to build
+			// the PR URL from the commit data we have available.
+			const pullRequest = {
+				url: buildPRURL( commit ),
+				creator: commit?.author?.login || 'unknown',
+			};
+
+			if ( pullRequest ) {
+				commitData.pullRequest = pullRequest;
+			}
+
+			return commitData;
+		} )
+	);
+
+	const processResult = pipe(
+		processCommits,
+		removeNesting,
+		dedupePRsPerLevel,
+		removeEmptyLevels,
+		sortLevels
+	);
+
+	console.log( `• Processing ${ commitsWithCommitData.length } commits` );
+	const result = processResult( commitsWithCommitData );
+
+	console.log( `• Generating Issue content` );
+	const content = generateIssueContent( result );
+
+	// Write the Markdown content to a file
+	fs.writeFileSync( nodePath.join( __dirname, 'issueContent.md' ), content );
+}
+
+function validateSince( sinceArg ) {
+	const maxMonths = 3;
```

Comment:
This'll need to be a bit bigger to accommodate the length of WP release cycles 😅

Reply:
What would you suggest? 4?

You have to provide the `--since` arg anyway (which @mikachan said should be set to date of last RC of previous major release) so this is just a safe guard to ensure that if someone accidentally provides a date far in the past we're not hitting up the API for thousands of commits 😓

Reply:
> include all PRs since the last plugin release to be fully included in Beta 1

You said this. So for this release, according to [this doc](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/versions-in-wordpress.md) that would be [16.7](https://github.com/WordPress/gutenberg/releases/tag/v16.7.0) which was 2023-09-23.

Am I right?

Reply:
Yes! 4 months should be ok.